### ISSUE TITLE
Offline banner

### ISSUE THREAD

**Issue by dansvend** [equinor/mad#374]
### Package / App

MAD Components

### Description

In mobile apps it's good to know why some functionality might not work as intended, and this could be due to being offline. Having a banner telling the user that the mobile device is disconnected from the internet is a good thing to have on most screens when this happens.

### Alternatives

1. Offline banner telling the user that the device is offline.

2. A toast telling that the user is offline which expands on press giving the user more details about what happened.

Either way I think this should be shown until the user dismisses it or it should not be dismissible at all.

**Comment by HaakonSvane** [comment]
I think this functionality would be nice to have in mad core. As I understand, mad navigation has support for custom banners which is used for service messages at the moment only. I think it would be nice to add this there instead of in mad components. Thoughts on this @dansvend  and @tormodAase  ?

**Comment by tormodAase** [comment]
I agree! We can put it at the top of the screen along with environment banner and service message

**Comment by dansvend** [comment]
That will be awesome! We can use [@react-native-community/netinfo with a Package Health Score of 89 on Snyk Advisor](https://snyk.io/advisor/npm-package/@react-native-community/netinfo).

```TSX
import { useNetInfo } from "@react-native-community/netinfo";
const { isConnected } = useNetInfo();
```

**Comment by dansvend** [comment]
> I think this functionality would be nice to have in mad core. As I understand, mad navigation has support for custom banners which is used for service messages at the moment only. I think it would be nice to add this there instead of in mad components. Thoughts on this @dansvend and @tormodAase ?

That will be awesome! It should still be added in MAD Components, and then use it from there and include it in mad-core in the custom headers, right?

We can use [@react-native-community/netinfo with a Package Health Score of 89 on Snyk Advisor](https://snyk.io/advisor/npm-package/@react-native-community/netinfo).

```TSX
import { useNetInfo } from "@react-native-community/netinfo";
const { isConnected } = useNetInfo();
```

**Comment by SUBBH-Equinor** [comment]
I've pushed changes for this ticket and assigned it to @Murugan-sudo  for review.

cc: @CaarnHeir
